synapse/storage/databases/main/purge_events.py

```diff
@@ -201,6 +202,14 @@ def _purge_history_txn(self, txn, room_id, token_str, delete_local_events):
         for event_id, _ in event_rows:
             txn.call_after(self._get_state_group_for_event.invalidate, (event_id,))
 
+        logger.info("[purge] removing events from destination_rooms")
+        txn.execute(
+            "DELETE FROM destination_rooms WHERE stream_ordering IN ("
+            "   SELECT stream_ordering FROM events_to_purge"
+            "       JOIN events USING (event_id)"
+            ")"
```

Comment:
destination_rooms doesn't have an index on `stream_ordering`, so this is going to be very inefficient.

Can we give `destination_rooms` an index on room_id?

Reply:
Following you about the missing index, but are you saying you want both indices on `stream_ordering` and `room_id`, or just the latter (and then to reword the query to use it)?

Reply:
just the latter. Hopefully it will need less index updating than an index on stream_ordering.

Reply:
btw: I think the `purge_room` api might also need an update (https://github.com/matrix-org/synapse/blob/master/docs/admin_api/purge_room.md)

Reply:
Is this better?